packages/components/src/List/Toolbar/ColumnChooser/ColumnDisplayer/ColumnDisplayer.component.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../../../../Icon';
+import ColumnOrder from './ColumnOrder';
+
+import theme from './ColumnDisplayer.scss';
+
+export const ColumnVisibility = ({ onChange, locked, value, t }) => {
+	if (locked) {
+		return <Icon name="talend-locked" />;
+	}
+	return (
+		<input
+			aria-label={t('CHECKBOX_VISIBILITY_COLUMN_CHOOSER', {
+				defaultValue: 'change visibility',
+			})}
+			className={classNames(
+				theme['tc-column-displayer-visibility-checkbox'],
+				'tc-column-displayer-visibility-checkbox',
+			)}
+			onChange={event => onChange(event, !value)}
+			type="checkbox"
+			checked={!value}
+			value={!value}
+		/>
+	);
+};
+
+ColumnVisibility.propTypes = {
+	locked: PropTypes.bool,
+	onChange: PropTypes.func.isRequired,
+	t: PropTypes.func.isRequired,
+	value: PropTypes.bool,
+};
+
+const ColumnDisplayer = ({
+	label,
+	hidden,
+	locked,
+	order,
+	length,
+	onChangeVisibility,
+	onBlurOrder,
+	onKeyPressOrder,
+	t,
+}) => (
+	<div
+		id="column-chooser-displayer"
```

Comment:
Please generate an id from a props.id.
This implies that it can't be used twice in a page.